File: DataStructure/88. Merge Sorted Array.cs
public class Solution {
        public void Merge(int[] nums1, int m, int[] nums2, int n) {
        int i = m - 1;
        int j = n - 1;
        int k = m + n - 1;
        
        while(i >= 0 && j >= 0){
            if(nums1[i] > nums2[j]){
                nums1[k] = nums1[i];
                i--;
            }else{
                nums1[k] = nums2[j];
                j--;
            }
            k--;
        }

        while(j >= 0){
            nums1[k] = nums2[j];
            k--;
            j--;
        }
    }
}

public class Solution {
    public void Merge(int[] nums1, int m, int[] nums2, int n) {
          for(var i = 0; i < nums2.Length; i++){
            nums1[nums1.Length-1-i] = nums2[i];
          }
          Array.Sort(nums1);
        }
}

File: DataStructure/88. Merge Sorted Array.py
# You are given two integer arrays nums1 and nums2, sorted in non-decreasing order, and two integers m and n, representing the number of elements in nums1 and nums2 respectively.
# Merge nums1 and nums2 into a single array sorted in non-decreasing order.
# The final sorted array should not be returned by the function, but instead be stored inside the array nums1. 
# To accommodate this, nums1 has a length of m + n, where the first m elements denote the elements that should be merged, and the last n elements are set to 0 and should be ignored. nums2 has a length of n.

nums1 = [1,2,3,0,0,0]
m = 3

nums2 = [2,5,6]
n = 3

class Solution:
    from bisect import bisect_right
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        """
        Do not return anything, modify nums1 in-place instead.
        """
        while nums1 and nums1[-1] == 0:
            nums1.pop()
        
        for num in nums2:
            index = bisect_right(nums1, num)
            nums1.insert(index, num)
        
        while len(nums1) < m + n :
            nums1.append(0)